docs/2-building-blocks-of-oop-part-1/hometask/src/Shape.ts

```diff
@@ -1,3 +1,46 @@
+import { Point } from './point';
+
 export abstract class Shape {
+  protected points: Point[];
+  protected color: string;
+  protected filled: boolean;
+
+  constructor(points: Point[]);
+  constructor(points: Point[], color: string, filled: boolean);
+
+  constructor(
+    points: Point[],
+    color: string = 'green',
+    filled: boolean = true
+  ) {
+    if (points.length < 3) {
+      throw new Error('At least 3 points are required.');
+    } else {
```

Comment:
We don't need an `else` statement here, because the `throw` stop the constructor execution

Reply:
removed else block